src/mkdocs_git_authors_plugin/git/page.py

```diff
@@ -238,18 +240,29 @@ def _process_git_log(self, sha, commit) -> None:
             --- (this method works through side effects)
         """
 
-        args = []
-        args.append("-1")  # Only existing sha
-        args.append(sha)
-        cmd = GitCommand("log", args)
-        cmd.run()
-
-        lines = cmd.stdout()
-
-        # in case of empty, non-committed files, raise error
-        if len(lines) == 0:
-            raise GitCommandError
-
+        co_authors = self._get_git_log(sha, commit)
+        for co_author in co_authors:
+            # Create the co-author
+            if co_author not in self._authors:
+                self._authors.append(co_author)
+            co_author.add_lines(self, commit)
+
+    def _get_git_log(self, sha, commit):
```

Comment:
Can we rename this function? It's only used for parsing co-authors?

Also, we should have already parsed all the commit earlier: 

https://github.com/timvink/mkdocs-git-authors-plugin/blob/370ee36afc3f0c7a2add374bec73862088cfc419/src/mkdocs_git_authors_plugin/git/repo.py#L116

It's better to extract and add  the co-author information in the Commit class:

https://github.com/timvink/mkdocs-git-authors-plugin/blob/master/src/mkdocs_git_authors_plugin/git/commit.py

then use it here. You won't need to have the caching in that case, and it should be even faster as you're not running `git log` again.

Reply:
Yeah I didn't read up enough, my bad. Your approach seems very reasonable. We're running `git blame`, and you added (a cached) `git_log` to get the co-author information. Either way, we're going to have to make a lot more `git` calls, which is a lot slower.

Maybe the better option is to not only have the caching, but also make processing co-authors optional (and disabled by default). The reason being is that the mainstream user flows will take a big performance hit to support a niche feature (git co-authors are not very common). 

Thoughts?